packages/node/src/integrations/tracing/nest.ts

```diff
@@ -44,7 +54,140 @@ interface MinimalNestJsApp {
 
 const INTEGRATION_NAME = 'Nest';
 
-export const instrumentNest = generateInstrumentOnce(INTEGRATION_NAME, () => new NestInstrumentation());
+const supportedVersions = ['>=8.0.0 <11'];
+
+const sentryPatched = Symbol('sentryPatched');
+
+/**
+ * Represents an injectable target class in NestJS.
+ */
+interface InjectableTarget {
+  name: string;
+  sentryPatched?: symbol;
+  prototype: {
+    use?: (req: unknown, res: unknown, next: (error?: Error | unknown) => void) => void;
+  };
+}
+
+/**
+ * Custom instrumentation for nestjs.
+ *
+ * This hooks into the @Injectable decorator, which is applied on class middleware, interceptors and guards.
+ */
+export class SentryNestInstrumentation extends InstrumentationBase {
+  public static readonly COMPONENT = '@nestjs/common';
+  public static readonly COMMON_ATTRIBUTES = {
+    component: SentryNestInstrumentation.COMPONENT,
+  };
+
+  public constructor(config: InstrumentationConfig = {}) {
+    super('sentry-nestjs', '1.0.0', config); // TODO: instrumentationVersion
+  }
+
+  /**
+   * Initializes the instrumentation by defining the modules to be patched.
+   */
+  public init(): InstrumentationNodeModuleDefinition {
+    const moduleDef = new InstrumentationNodeModuleDefinition(SentryNestInstrumentation.COMPONENT, supportedVersions);
+
+    moduleDef.files.push(this._getInjectableFileInstrumentation(supportedVersions));
+    return moduleDef;
+  }
+
+  /**
+   * Wraps the @Injectable decorator.
+   */
+  private _getInjectableFileInstrumentation(versions: string[]): InstrumentationNodeModuleFile {
+    return new InstrumentationNodeModuleFile(
+      '@nestjs/common/decorators/core/injectable.decorator.js',
+      versions,
+      (moduleExports: { Injectable: InjectableTarget }) => {
+        if (isWrapped(moduleExports.Injectable)) {
+          this._unwrap(moduleExports, 'Injectable');
+        }
+        this._wrap(moduleExports, 'Injectable', this._createWrapInjectable());
+        return moduleExports;
+      },
+      (moduleExports: { Injectable: InjectableTarget }) => {
+        this._unwrap(moduleExports, 'Injectable');
+      },
+    );
+  }
+
+  /**
+   * Creates a wrapper function for the @Injectable decorator.
+   *
+   * Wraps the use method to instrument nest class middleware.
+   */
+  private _createWrapInjectable() {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return function wrapInjectable(original: any) {
+      return function wrappedInjectable(options?: unknown) {
+        return function (target: InjectableTarget) {
+          // ensure class has not been patched before
+          if (target.sentryPatched) {
```

Comment:
should be moved inside the `if (typeof target.prototype.use)` so that we only attach this property for actually patched injectables